dftimewolf/lib/collectors/virustotal.py

```diff
@@ -0,0 +1,231 @@
+        self.state.StoreContainer(container)
+        self.logger.info('Finished writing evtx to file')
+
+  # pylint: disable=arguments-differ,too-many-arguments
+  def SetUp(
+      self,
+      hashes: str,
+      vt_api_key: str,
+      vt_type: str,
+      output_path: str = tempfile.mkdtemp(),
+  ) -> None:
+    """Sets up an Virustotal (VT) collector.
+
+    Args:
+      hashes: Coma seperated strings of hashes
+      vt_api_key: Virustotal Enterprise API Key
+      vt_type: Which file to fetch
+      output_path [optional]: Where to store the downloaded files to
+    """
+
+    self.output_path = self._CheckOutputPath(output_path)
+
+    if not hashes:
+      self.ModuleError("You need to specify at least one hash", critical=True)
+      return
+
+    if not vt_type:
+      self.ModuleError(
+          "You need to specify an vt_type from: pcap, evtx", critical=True)
+      return
+
+    self.vt_type = vt_type
+
+    self.hashes_list = [item.strip() for item in hashes.strip().split(",")]
+
+    if not vt_api_key:
+      self.ModuleError(
+          "You need to specify a Virustotal Enterprise API key",
+          critical=True,
+      )
+      return
+
+    self.client = vt.Client(vt_api_key)
+
+    if self.client is None:
+        self.ModuleError(
+            f'Error creating Virustotal Client instance',
+            critical=True,
+        )
+        return
+
+  def _CheckOutputPath(self, output_path: str = tempfile.mkdtemp()) -> str:
+    """Checks that the output path can be manipulated by the module.
+
+    Args:
+    output_path: Full path to the output directory where files will be
+        dumped.
+
+    Returns:
+        The full path to the directory where files will be dumped.
+    """
+    # Check that the output path can be manipulated
+    if not output_path:
+      return tempfile.mkdtemp()
+    elif os.path.exists(output_path):
+      return output_path
+    elif not os.path.exists(output_path):
+      try:
+        os.makedirs(output_path)
+        return output_path
+      except OSError as error:
+        self.ModuleError(
+            "{0:s} error while creating the output directory: {1!s}".format(
+                output_path, error),
+            critical=True,
+        )
+        """Below should never be reached, but Either all return statements in 
+        a function should return an expression, or none of them should. 
+        (inconsistent-return-statements)
+        """
+        return tempfile.mkdtemp()
+    elif not os.path.isdir(output_path):
+      self.ModuleError(output_path + " is not a directory:", critical=True)
+      return tempfile.mkdtemp()
+    else:
+      return tempfile.mkdtemp()
+
+  def _isHashKnownToVT(self, vt_hash: str) -> bool:
+    """Checks if a hash is known to VT.
+
+          Args:
+              vt_hash ([str]): A hash.
+
+          Returns:
+              Bool: True if found on VT
+              False: File not found on VT.
+          """
```

Comment:
Please fix the indentation here. 

Also, no need for brackets in the type definition here. Actually, you can also remove type definitions from the docstrings in these files given that you're already giving them as type hints